oauth_tools/external_idp.py

```diff
@@ -147,23 +150,28 @@ def _apply_dex_resources(self) -> None:
             self._client.apply(obj, force=True)
 
         logger.info("Restarting dex")
-        self._restart_dex()
+        deleted_pod_names = self._restart_dex()
 
         logger.info("Waiting for dex to be ready")
-        self._wait_until_is_ready()
-
-    def __wait_until_is_ready(self) -> None:
-        for pod in self._client.list(Pod, namespace=self.namespace, labels={"app": "dex"}):
-            # Some pods may be deleted, if we are restarting
-            try:
+        self._wait_until_is_ready(ignore=deleted_pod_names)
+
+    def __wait_until_is_ready(self, ignore: Optional[List[str]] = None) -> None:
+        if ignore is None:
+            ignore = []
+        ready = False
```

Comment:
Just have a question here (I might not fully understand it yet): why do we need this extra flag? We have already ignored the deleted pods in the `for` loop, right?

Reply:
I think this is the same problem discussed in here: https://github.com/canonical/iam-bundle/pull/266#discussion_r1722351260